typescript/sdk/src/aws/s3.ts

```diff
@@ -17,37 +17,60 @@ export class S3Wrapper {
   readonly region: string;
   readonly folder: string | undefined;
 
+  private cache: Record<string, S3Receipt<any>> | undefined;
+
   static fromBucketUrl(bucketUrl: string): S3Wrapper {
     const match = bucketUrl.match(S3_BUCKET_REGEX);
     if (!match) throw new Error('Could not parse bucket url');
     return new S3Wrapper(match[1], match[2], undefined);
   }
 
-  constructor(bucket: string, region: string, folder: string | undefined) {
+  constructor(
+    bucket: string,
+    region: string,
+    folder: string | undefined,
+    caching = false,
```

Comment:
These kind of things I'd personally prefer as an object (as it becomes harder to understand at the call-site)

Reply:
yeah FWIW i just copied this from infra but will convert to object like @jmrossy suggested